yarn-project/end-to-end/src/deploy_l1_contracts.ts

```diff
@@ -1,23 +1,70 @@
-import { EthereumRpc } from '@aztec/ethereum.js/eth_rpc';
-import { WalletProvider } from '@aztec/ethereum.js/provider';
-import { Rollup, UnverifiedDataEmitter } from '@aztec/l1-contracts';
-
-export const deployRollupContract = async (provider: WalletProvider, ethRpc: EthereumRpc) => {
-  const deployAccount = provider.getAccount(0);
-  const contract = new Rollup(ethRpc, undefined, { from: deployAccount, gas: 1000000 });
-  await contract.deploy().send().getReceipt();
-  return contract.address;
-};
+import { DebugLogger, EthAddress } from '@aztec/foundation';
+import {
+  RollupAbi,
+  RollupBytecode,
+  UnverifiedDataEmitterAbi,
+  UnverifiedDataEmitterBytecode,
+} from '@aztec/l1-contracts/viem';
+import type { Abi, Narrow } from 'abitype';
```

Comment:
what does `Narrow` do, can't find documentation on `abitype`'s website

Reply:
<img width="1239" alt="image" src="https://user-images.githubusercontent.com/13470840/234822775-9d8233a8-f5d5-4401-8c15-bdb9cb88ea63.png">

Yes, the types are not the easiest to comprehend 😆